2025/lab/Keys/Week6_Interaction Terms.Rmd

```diff
@@ -0,0 +1,299 @@
+
+Interpreting the coefficients of interaction terms directly from the regression output can be non-intuitive because the coefficients of the main effects and the interaction term need to be considered together. This is where functions like `predict` and packages like `marginaleffects` or `margins` become extremely useful. We touched upon predict in the past but will cover more of it and marginal effects in a future lab. 
+
+## Dichotomous Variable X Multinominal Variable (more than two categories)
+
+We can create and interpret an interaction model with self-employment status (dichotomous) and political party affiliation (multinominal).
+```{r dum-cat}
+dum_cat_base <- lm(income ~ self_employed + partyid,
+                   data = gss)
+dum_cat_int <- lm(income ~ self_employed * partyid,
+                  data = gss)
+
+modelsummary(list(dum_cat_base, dum_cat_int),
+             estimate = "{estimate}{stars}",
+             fmt = 3,
+             statistic = NULL,
+             gof_omit = "IC|Log|alg|pss|F|RMSE",
+             notes = "Notes: * p < 0.05, ** p < 0.01, *** p < 0.001",
+             output="huxtable")
+```
+
+
+## Dummy X Continuous
+
+We can also create interactions between dummy variables and continuous variables. In the example below, we can hypothesize that income is affected by the hours a person works and whether they have a college degree or higher.
+
+```{r dum-con}
+gss <- gss %>% 
+  mutate(college = case_when(
+    degree %in% c("Bach", "Grad") ~ 1,  # College or Grad
+    degree %in% c("Less_HS", "HS", "Assoc") ~ 0   # Non-college
+  ))
+# table(gss$college, gss$degree)
+
+dum_con_base <- lm(income ~ workhrs + college,
+                   data = gss)
+dum_con_int <- lm(income ~ workhrs * college,
+                  data = gss)
+
+modelsummary(list(dum_con_base, dum_con_int),
+             estimate = "{estimate}",
+             fmt = 3,
+             statistic = NULL,
+             gof_omit = "IC|Log|alg|pss|F|RMSE",
+             notes = "Notes: * p < 0.05, ** p < 0.01, *** p < 0.001",
+             output="huxtable")
+```
+
+
+## Continuous X Continuous
+
+Rounding out the simple cases of interactions, we have continuous X continuous interactions.
+```{r con-con}
+con_con_base <- lm(income ~ workhrs + educ,
+                   data = gss)
+con_con_int <- lm(income ~ workhrs * educ,
+                  data = gss)
+
+modelsummary(list(con_con_base, con_con_int),
+             estimate = "{estimate}",
+             fmt = 3,
+             statistic = NULL,
+             gof_omit = "IC|Log|alg|pss|F|RMSE",
+             notes = "Notes: * p < 0.05, ** p < 0.01, *** p < 0.001",
+             output="huxtable")
+```
+
+## Interactions Between Multiple Variables
+You aren't limited to only interacting two variables. You could have 3-way interactions, 4-way interactions, etc. If you find that you need to do this, don't hesitate to reach out. 
```

Comment:
This is fine, but I'd add that these models become difficult to interpret and should be used with extreme caution. And rarely would we want to implement a 4-way interaction.